packages/@aws-cdk/aws-codebuild/lib/build-spec.ts

```diff
@@ -70,6 +80,22 @@ class ObjectBuildSpec extends BuildSpec {
   }
 }
 
+/**
+ * BuildSpec that exports into YAML format
+ */
+class YamlBuildSpec extends BuildSpec {
+  public readonly isImmediate: boolean = true;
+
+  constructor(public readonly spec: {[key: string]: any}) {
+    super();
+  }
+  public toBuildSpec(): string {
+    return Lazy.uncachedString({
```

Comment:
Any reason this uses a Lazy? Can't we just go

```ts
  public toBuildSpec(): string {
    return yaml_cfn.serialize(this.spec);
  }
```

?

Reply:
sure, it worked fine, initially I based on the impl of `fromObject` 

I thought it could have something